vllm/worker/habana_model_runner.py

```diff
@@ -147,45 +151,62 @@ def align_workers(value, op):
     return value_t.item()
 
 
+def pad_list(l, k, v):
+    target_len = round_up(len(l), k)
+    padding = target_len - len(l)
+    return l + [v] * padding
+
+
 class HpuModelAdapter():
 
-    def __init__(self, model, enforce_eager):
+    def __init__(self, model, block_size, enforce_eager):
         self.model = model
+        self.block_size = block_size
         if not htorch.utils.internal.is_lazy() and not enforce_eager:
             self.model = torch.compile(self.model,
                                        backend='hpu_backend',
                                        dynamic=False)
 
-    def _set_attn_bias(self, attn_metadata, batch_size, seq_len, device,
-                       dtype):
-        prefill_metadata = attn_metadata
-        if prefill_metadata is None:
-            return attn_metadata
-
-        seq_lens_t = prefill_metadata.seq_lens_tensor
-        len_mask = (torch.arange(0, seq_len, device=device,
-                                 dtype=torch.int32).view(1, seq_len).ge(
-                                     seq_lens_t.unsqueeze(-1)).view(
-                                         batch_size, 1, 1, seq_len))
-        causal_mask = torch.triu(torch.ones((batch_size, 1, seq_len, seq_len),
-                                            device=device,
-                                            dtype=torch.bool),
-                                 diagonal=1)
+    def _set_attn_bias(self, metadata, batch_size, seq_len, device, dtype):
+        seq_lens_t = metadata.seq_lens_tensor
+        len_mask = (torch.arange(0, seq_len, device=device, dtype=torch.int32)
+                    .view(1, seq_len)
+                    .ge(seq_lens_t.unsqueeze(-1))
+                    .view(batch_size, 1, 1, seq_len))
+        causal_mask = torch.triu(
+            torch.ones((batch_size, 1, seq_len, seq_len), device=device, dtype=torch.bool),
+            diagonal=1
+        )
         mask = causal_mask.logical_or(len_mask)
-        attn_bias = (torch.zeros_like(mask, dtype=dtype).masked_fill_(
-            mask, -math.inf))
-        #FIXME: Restore sliding window support
-        #if self.sliding_window is not None:
-        attn_metadata = prefill_metadata._replace(attn_bias=attn_bias)
+        attn_bias = (torch.zeros_like(mask, dtype=dtype)
+                        .masked_fill_(mask, -math.inf))
+        return metadata._replace(attn_bias=attn_bias)
+
+    def _set_block_mapping(self, metadata, batch_size, device, dtype):
+        mask = torch.arange(0, self.block_size, device=device, dtype=torch.int32).unsqueeze(0)
+        mask = mask >= metadata.block_usage.unsqueeze(-1)
+        attn_bias = (torch.zeros_like(mask, dtype=dtype)
+                        .masked_fill_(mask, -math.inf))
+        block_mapping = torch.nn.functional.one_hot(metadata.block_mapping, num_classes=batch_size).to(dtype)
```

Comment:
the input to one_hot should be cast to torch.long, otherwise, it has runtime error